Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -1418,19 +1421,9 @@ class Gx2Fitter {
         break;
       }
 
-      if (multipleScattering) {
-        // update the scattering angles
-        for (std::size_t matSurface = 0; matSurface < nMaterialSurfaces;
-             matSurface++) {
-          const std::size_t deltaPosition = eBoundSize + 2 * matSurface;
-          const GeometryIdentifier geoId = geoIdVector[matSurface];
-          const auto scatteringMapId = scatteringMap.find(geoId);
-          assert(scatteringMapId != scatteringMap.end() &&
-                 "No scattering angles found for material surface.");
-          scatteringMapId->second.scatteringAngles().block<2, 1>(2, 0) +=
-              deltaParamsExtended.block<2, 1>(deltaPosition, 0).eval();
-        }
-      }
+      updateGx2fParams(params, deltaParamsExtended, nMaterialSurfaces,
+                       scatteringMap, geoIdVector);
+      ACTS_VERBOSE("Updated parameters: " << params.parameters().transpose());
```

Comment:
is the
```c++
if (multipleScattering) {
```
guarding not needed anymore?

Reply:
The information is also contained in `nMaterialSurfaces`, resulting in a 0-iteration loop:

```
for (std::size_t matSurface = 0; matSurface < nMaterialSurfaces;
       matSurface++) {
```